rtt/transports/mqueue/MQSendRecv.cpp

```diff
@@ -75,11 +76,14 @@ void MQSendRecv::setupStream(base::DataSourceBase::shared_ptr ds, base::PortInte
     marshaller_cookie = mtransport.createCookie();
     mis_sender = is_sender;
 
-    std::stringstream namestr;
-    namestr << '/' << port->getInterface()->getOwner()->getName() << '.' << port->getName() << '.' << this << '@' << getpid();
-
     if (policy.name_id.empty())
-        policy.name_id = namestr.str();
+    {
+        std::stringstream name_stream;
+        name_stream << port->getInterface()->getOwner()->getName() << '.' << port->getName() << '.' << this << '@' << getpid();
```

Comment:
I am not familiar with the naming requirements of message queues as I never used that transport myself. If you say it has to be globally unique, whatever global means in this context, it might be the best option to log an error that explains the issue and throw. I leave that decision to you. But we should definitely avoid segmentation faults and I would ask you to implement whatever exception behavior you find most reasonable. Or add a new issue with a reference to this discussion so that we at least not forget about it later.